File: app/src/main/java/cn/intret/app/picgo/model/CompareItem.java
package cn.intret.app.picgo.model;


import java.io.File;

/**
 * 比较项。在文件移动冲突比较中每个一个比较的项目，这个比较会有一个结果决定选择保留左边还是右边。
 */
public class CompareItem {
    ResolveResult mResult;
    File mTargetFile;
    File mSourceFile;

    public ResolveResult getResult() {
        return mResult;
    }

    public CompareItem setResult(ResolveResult result) {
        mResult = result;
        return this;
    }

    public File getTargetFile() {
        return mTargetFile;
    }

    public CompareItem setTargetFile(File targetFile) {
        mTargetFile = targetFile;
        return this;
    }

    public File getSourceFile() {
        return mSourceFile;
    }

    public CompareItem setSourceFile(File sourceFile) {
        mSourceFile = sourceFile;
        return this;
    }

    @Override
    public String toString() {
        return "CompareItem{" +
                "mResult=" + mResult +
                ", mTargetFile=" + mTargetFile +
                ", mSourceFile=" + mSourceFile +
                '}';
    }
}

File: app/src/main/java/cn/intret/app/picgo/model/image/CompareItem.kt
package cn.intret.app.picgo.model.image


import java.io.File

/**
 * 比较项。在文件移动冲突比较中每个一个比较的项目，这个比较会有一个结果决定选择保留左边还是右边。
 */
class CompareItem {
    internal lateinit var result: ResolveResult
    internal lateinit var targetFile: File
    internal lateinit var sourceFile: File

    constructor(result: ResolveResult, targetFile: File, sourceFile: File) {
        this.result = result
        this.targetFile = targetFile
        this.sourceFile = sourceFile
    }


    override fun toString(): String {
        return "CompareItem{" +
                "result=" + result +
                ", targetFile=" + targetFile +
                ", sourceFile=" + sourceFile +
                '}'.toString()
    }
}
